plugins/catalog-backend/src/ingestion/processors/GithubDiscoveryProcessor.ts

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { LocationSpec } from '@backstage/catalog-model';
+import { Config } from '@backstage/config';
+import {
+  GithubCredentialsProvider,
+  GitHubIntegrationConfig,
+  readGitHubIntegrationConfigs,
+} from '@backstage/integration';
+import { graphql } from '@octokit/graphql';
+import { Logger } from 'winston';
+import { getOrganizationRepositories } from './github';
+import * as results from './results';
+import { CatalogProcessor, CatalogProcessorEmit } from './types';
+
+/**
+ * Extracts repositories out of a GitHub org.
+ */
+export class GithubDiscoveryProcessor implements CatalogProcessor {
+  private readonly gitHubConfigMap: Map<string, GitHubIntegrationConfig>;
+  private readonly logger: Logger;
+
+  static fromConfig(config: Config, options: { logger: Logger }) {
+    const configs = readGitHubIntegrationConfigs(
```

Comment:
i think you could benefit a lot from using the full power of the `integration` package here. Something like

```ts
const integrations = ScmIntegrations.fromConfig(config);
// pass that ^ into the constructor, and then later, below
this.integrations.github.byUrl(location.target) // <-- has .config etc
```